samples/bluetooth/mesh_provisioner/src/main.c

```diff
@@ -294,6 +298,24 @@ static uint8_t check_unconfigured(struct bt_mesh_cdb_node *node, void *data)
 	return BT_MESH_CDB_ITER_CONTINUE;
 }
 
+static const struct gpio_dt_spec button = GPIO_DT_SPEC_GET_OR(SW0_NODE, gpios, {0});
+static struct gpio_callback button_cb_data;
+
+static void timeout_timer_expired(struct k_timer *timer)
+{
+	printk("Provisioner timed out, press button at %s pin %d to continue.\n", button.port->name,
+	       button.pin);
+}
+
+K_TIMER_DEFINE(timeout_timer, timeout_timer_expired, NULL);
```

Comment:
Might sound a little bit strange, but in general we try to use `k_work_delayable` rather than timers. Could you try to duplicate the behaviour using `k_work_delayable`?